pkg/cmd/build/log_filter.go

```diff
@@ -0,0 +1,79 @@
+package build
+
+import (
+	"fmt"
+	"github.com/moby/buildkit/client"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"regexp"
+)
+
+type ConditionFunc func(vertex *client.Vertex, ss *client.SolveStatus) bool
+type TransformerFunc func(vertex *client.Vertex, ss *client.SolveStatus, progress string)
+
+type Rule struct {
+	condition   ConditionFunc
+	transformer TransformerFunc
+}
+
+type BuildKitLogsFilter struct {
+	rules []Rule
+}
+
+func NewBuildKitLogsFilter(rules []Rule) *BuildKitLogsFilter {
+	return &BuildKitLogsFilter{
+		rules: rules,
+	}
+}
+
+func (lf *BuildKitLogsFilter) Run(ss *client.SolveStatus, progress string) {
+	for _, vertex := range ss.Vertexes {
+		for _, rule := range lf.rules {
+			if rule.condition(vertex, ss) {
+				rule.transformer(vertex, ss, progress)
```

Comment:
```suggestion
				oktetoLog.Info("could not detect cache image")
				rule.transformer(vertex, ss, progress)
```

Reply:
@jLopezbarb the code is generic but the error log would be specific to the cache image. We could move it to the `BuildKitMissingCacheTransformer` func but I think it's already available in logs the message: `[skip] cache image not available: %s`. Do you think we need an extra log? Let me know!

Reply:
log added!